pyphare/pyphare/pharesee/hierarchy/fromh5.py

```diff
@@ -0,0 +1,320 @@
+import os
+import numpy as np
+
+from .patch import Patch
+from .patchlevel import PatchLevel
+from .patchdata import FieldData, ParticleData
+from ..particles import Particles
+from .hierarchy import PatchHierarchy
+from .hierarchy import format_timestamp
+from ...core.box import Box
+from ...core.phare_utilities import (
+    refinement_ratio,
+    none_iterable,
+    all_iterables,
+)
+from ...core.gridlayout import GridLayout
+from .hierarchy_utils import field_qties
+import h5py
+from pathlib import Path
+from pyphare.core.phare_utilities import listify
+
+
+h5_time_grp_key = "t"
+particle_files_patterns = ("domain", "patchGhost", "levelGhost")
+
+
+def get_all_available_quantities_from_h5(filepath, time=0, exclude=["tags"]):
+    time = format_timestamp(time)
+    hier = None
+    path = Path(filepath)
+    for h5 in path.glob("*.h5"):
+        if h5.parent == path and h5.stem not in exclude:
+            hier = hierarchy_fromh5(str(h5), time, hier)
+    return hier
```

Comment:
explain how it is is `mutable`, given it's not changed in the current scope?

Your suggestion, is changing it, so it would have to be `mutable` in some sense